### PR TITLE
Adding ML Pipelines WG charter

### PR DESCRIPTION
**Pull request by paveldournov**
ML Pipelines workgroup charter. Initial commit.

### CODE REVIEW

**Comment by rmgogogo** [comment]
/hold to give Thea a review.
Feel free to submit after Thea OK with it

**Comment by jlewi** [comment]
Pinging the other WG leads to take a look;

@animeshsingh 
@cliveseldon 
@yuzisun 
@ellistarn 
@neuromage 
@paveldournov 
@elikatsis
@vpavlin 
@yanniszark 
@Jeffwan 
@krishnadurai 
@terrytangyuan 
@gaocegege 
@johnugeorge 
@aronchick 
@StefanoFioravanzo 
@jbottum 
@elviraux|  
@kimwnasptd 
@krazyhaas

File: wgs/wg-pipelines/wg-charter-pipelines.md
Comment (terrytangyuan) [inline wgs/wg-pipelines/wg-charter-pipelines.md:78]: Where's the list of chairs and leads?

File: wgs/wg-pipelines/wg-charter-pipelines.md
Comment (terrytangyuan) [inline wgs/wg-pipelines/wg-charter-pipelines.md:87]: Change YOURWG to name of this WG

File: wgs/wg-pipelines/wg-charter-pipelines.md
Comment (animeshsingh) [inline wgs/wg-pipelines/wg-charter-pipelines.md:57]:
```suggestion
- KFP Tekton Backend (fork, with long term intent of merging in mainstream pipelines repo)
```

File: wgs/wg-pipelines/wg-charter-pipelines.md
Comment (animeshsingh) [inline wgs/wg-pipelines/wg-charter-pipelines.md:71]:
```suggestion
- Feature and roadmap planning
- Integration process with other external projects like Kale etc.
- Support community for investigating KFP plug-points e.g. pluggable advanced scheduler for KFP to support gang scheduling, job priority, pluggable external metadata engines etc.
```

File: wgs/wg-pipelines/wg-charter-pipelines.md
Comment (animeshsingh) [reply wgs/wg-pipelines/wg-charter-pipelines.md:78]: +1 on above.

**Comment by jlewi** [comment]
Per kubeflow/metadata#225 is that going to be one of the repositories owned by this WG?

File: wgs/wg-pipelines/wg-charter-pipelines.md
Comment (neuromage) [reply wgs/wg-pipelines/wg-charter-pipelines.md:71]: @animeshsingh what do you mean by "pluggable external metadata engines"?

**Comment by jlewi** [comment]
@paveldournov ping?

File: wgs/wg-pipelines/wg-charter-pipelines.md
Comment (paveldournov) [reply wgs/wg-pipelines/wg-charter-pipelines.md:57]: Agreed.

File: wgs/wg-pipelines/wg-charter-pipelines.md
Comment (paveldournov) [reply wgs/wg-pipelines/wg-charter-pipelines.md:71]: This is a good point, though I'd not make it so specific in the WG. We should support the community to create extensibility points in the ML Pipelines stack and support development of the extensions.

Comment (paveldournov) [reply wgs/wg-pipelines/wg-charter-pipelines.md:71]: @animeshsingh please see the added text below

**Comment by jlewi** [comment]
We're missing an entry in
https://github.com/kubeflow/community/blob/master/wgs.yaml

File: wgs/wg-pipelines/wg-charter-pipelines.md
Comment (animeshsingh) [reply wgs/wg-pipelines/wg-charter-pipelines.md:78]: thanks @paveldournov 

except for the wg.yaml, fine with this.

**Review by theadactyl — APPROVED** [review]

**Comment by theadactyl** [comment]
@jlewi @animeshsingh can you comment with `@googlebot I consent` to unblock CLA check?

File: wgs/wg-pipelines/wg-charter-pipelines.md
Comment (theadactyl) [reply wgs/wg-pipelines/wg-charter-pipelines.md:87]: Being added in the wgs.yaml here: https://github.com/kubeflow/community/pull/395/commits/eb2c2dd8fb809f133d7cf20c8775c7804aeffc8b

**Comment by jlewi** [comment]
@googlebot I consent

File: wgs/wg-pipelines/wg-charter-pipelines.md
Comment (andreyvelich) [inline wgs/wg-pipelines/wg-charter-pipelines.md:54]: Can we also add AutoML dependency?

```suggestion
- TF Job and other training controllers
- Kubeflow AutoML experiments
```

File: wgs.yaml
Comment (Jeffwan) [inline wgs.yaml:110]: I feel like it's better to have some chairs or leads from this project.

File: wgs/wg-pipelines/wg-charter-pipelines.md
Comment (Jeffwan) [inline wgs/wg-pipelines/wg-charter-pipelines.md:50]: Could you explain the dependencies here? Are you talking about reusable component?

**Comment by jlewi** [comment]
cc @kubeflow/wg-automl-leads @kubeflow/wg-serving-leads @kubeflow/wg-training-leads

File: wgs/wg-pipelines/wg-charter-pipelines.md
Comment (paveldournov) [reply wgs/wg-pipelines/wg-charter-pipelines.md:50]: These mean to represent dependencies that some users of ML Pipeline may need to complete their scenarios. For example, the multi-user support require the profile controller, the pipelines that run distributed training in the cluster depend on the TFJob controller, etc. The dependencies are not hard, so the ML Pipelines can function without them.

File: wgs/wg-pipelines/wg-charter-pipelines.md
Comment (paveldournov) [reply wgs/wg-pipelines/wg-charter-pipelines.md:54]: In what scenarios this dependency would materialize?

File: wgs.yaml
Comment (rmgogogo) [reply wgs.yaml:110]: Jeff means "kfp-tekton". Following the structure, I updated Pavel's PR to add Animesh to owners list for easy contacting.
https://github.com/kubeflow/community/blob/3e8bb28c49f872bd8395a7c5887b6516bbaab14c/generator/app.go#L100

"make generate" to refresh the auto-generated markdown file will be handled later after LGTM.

File: wgs/wg-pipelines/wg-charter-pipelines.md
Comment (rmgogogo) [reply wgs/wg-pipelines/wg-charter-pipelines.md:78]: Thanks Animesh. So here mark this part is resolved.

File: wgs/wg-pipelines/wg-charter-pipelines.md
Comment (rmgogogo) [reply wgs/wg-pipelines/wg-charter-pipelines.md:50]: updated the markdown here to detail the scenario

File: wgs/wg-pipelines/wg-charter-pipelines.md
Comment (rmgogogo) [reply wgs/wg-pipelines/wg-charter-pipelines.md:57]: I think Andrey means Katib, right?
https://github.com/kubeflow/pipelines/pull/754/files

By the way, we also have other options for HP tune etc., e.x. in-trainer Keras Tunner, out-trainer TFX Tunner component, [Google Cloud HP Tuning](https://cloud.google.com/ai-platform/training/docs/using-hyperparameter-tuning), [Google Cloud AutoML](https://cloud.google.com/automl) etc. 

Here just list the other Kubeflow components per the context.

**Comment by rmgogogo** [comment]
Helped Pavel to updated  and resolved few comments. Please take a review.

I will help generate the files via "make generate" later before the submit.

File: wgs/wg-pipelines/wg-charter-pipelines.md
Comment (andreyvelich) [reply wgs/wg-pipelines/wg-charter-pipelines.md:57]: Yes, thanks @rmgogogo.
This launcher is a bit outdated, we will update launcher with new v1beta1 version after we release it.

File: wgs.yaml
Comment (animeshsingh) [reply wgs.yaml:110]: thanks @rmgogogo 
Additionally, since the overall leadership should reflect the leadership from sub-projects, it would be good have a representation either in chairs or leads section at top.

**Comment by animeshsingh** [comment]
@googlebot I consent

File: wgs/wg-pipelines/wg-charter-pipelines.md
Comment (paveldournov) [reply wgs/wg-pipelines/wg-charter-pipelines.md:98]: It is taken from the #1 option in https://github.com/paveldournov/community/blob/master/wgs/wg-governance.md#subproject-creation
The original doc makes it a bit ambiguous, so changing it to WG Leads (TLs and Chairs).

File: wgs.yaml
Comment (paveldournov) [reply wgs.yaml:113]: done

**Comment by paveldournov** [comment]
@googlebot I consent

File: wgs.yaml
Comment (rmgogogo) [reply wgs.yaml:113]: so here mark as resolved. I will help take care this PR for better format, submit etc. The people list will be shorted by alphabet order per the tool.

**Comment by elikatsis** [comment]
Hi everyone!

I'd like to be included as a tech lead in this WG on behalf of Arrikto for the following reasons:
1. We have made contributions to the core DSL, most of them targeted to volume handling, and we are planning to do mroe
2. We are heavily working on KFP integration with the rest of Kubeflow components like Katib, KFService, Metadata, to be able to showcase e2e workflows with Kale
3. We have active contributions as far as multi-user KFP are concerned both in terms of design and implementation, with more to come

Essentially, those are the reasons why I was included in the initial proposal as well.

Thanks!

**Comment by Ark-kun** [comment]
Should we make sure that all KF Pipelines owners are included in the WG Pipelines Leads list? https://github.com/kubeflow/pipelines/blob/master/OWNERS Also see the KFP contributors https://github.com/kubeflow/pipelines/graphs/contributors

We can also include more people who're actively involved with the community e.g. @numerology 

What do you think?

**Comment by paveldournov** [comment]
All these are great contributions and we are not proposing changes to the OWNERS file and encourage further contributions to the core code, designs, and additional tooling within and around ML Pipelines.

The purpose of tech leads is to ensure that the project(s) sponsored by the WG have the technical leadership and maintainer resources needed, not simply to recognize individual efforts in the community. We want to follow the recommendation to have no more than 6 leads (currently at 7) to ensure effective decision-making.

This is a new process and we need to give it some time to settle and establish the process of roadmap setting, feature planning, and execution. We can reassess the the mechanisms to grant additional responsibility to WG members and will revisit the leads membership based on project needs in the future.

File: OWNERS_ALIASES
Comment (jlewi) [inline OWNERS_ALIASES:13]: Fix @kart
```suggestion
  wg-pipelines-leads:
    - Bobgy
    - animeshsingh
    - james-jwu
    - kramachandran
    - neuromage
    - paveldournov
    - rmgogogo
```

**Comment by animeshsingh** [comment]
@kramachandran needs to be part of the Kubeflow org

**Comment by jlewi** [comment]
@animeshsingh Yup the github sync was broken kubeflow/internal-acls#344

**Comment by animeshsingh** [comment]
@kramachandran you would have gotten an email associated with your github to accept org membership. Can you please do that so that we can move this forward.

**Comment by elikatsis** [comment]
OK, this is completely understood. However from a corresponding discussion in the Notebooks WG, Thea proposed that it is good to have a bit more diversity in the WGs and my understanding is that currently the proposal for the KFP WG is to have 6 people from Google and Animesh from IBM. Since we, at Arrikto, are committed long term to KFP and heavily invested, it may make sense for a bit more diversity in this WG.
But I'll let you decide.

**Comment by jlewi** [comment]
I believe these are the right group of leads.

As I [mentioned](https://github.com/kubeflow/community/pull/407#discussion_r490216128) on the notebooks PR, representation of multiple organizations is desirable but not my primary concern. My expectation is that the leads are the ones accountable for the success or failures of the projects. 

@kramachandran is out for a week or so. I will merge this PR manually.

@Bobgy @rmgogogo @james-jwu @animeshsingh could one of you please open a PR to create kubeflow/wg-pipeline-leads

**Pull request merged**
